packages/components/pie-cookie-banner/src/defs.ts

```diff
@@ -63,7 +63,13 @@ export interface CookieBannerProps {
      */
     cookieTechnologiesLink: string;
 
-    defaultPreferences: Record<PreferenceIds, boolean> | object
+    /**
+     * Allows consumers to pass in specific preference(s) to the component which will toggle
+     * the switch to be on by default (if set to `true`).
+     *
+     * e.g. { 'functional': true }
+     */
+    defaultPreferences: Record<PreferenceIds, boolean> | NonNullable<unknown>;
```

Comment:
I belive this prop (along with a few others) needs to be optional as well. Otherwise there are issues with Typescript React consumers